lib/warnings.js

```diff
@@ -27,6 +27,8 @@ warning.create('FastifyDeprecation', 'FSTDEP013', 'Direct return of "trailers" f
 
 warning.create('FastifyDeprecation', 'FSTDEP014', 'You are trying to set/access the default route. This property is deprecated. Please, use setNotFoundHandler if you want to custom a 404 handler or the wildcard (*) to match all routes.')
 
+warning.create('Fastify Warning', 'FSTWARN001', 'The secondary server creation is being attempted')
```

Comment:
```suggestion
warning.create('Fastify Warning', 'FSTWARN002', 'Secondary server is already listening')
```

wdyt?

Reply:
For some reason I get this error
```log
 FAIL ​ test/schema-feature.test.js
 ✖ test unfinished

  test/schema-feature.test.js                                                  
  255 | })                                                                     
  256 |                                                                        
> 257 | test('Should emit warning if the schema headers is undefined', t => {  
      | ^                                                                      
  258 |   t.plan(4)                                                            
  259 |   const fastify = Fastify()                                            
  260 |                                                                        

  test: Should emit warning if the schema headers is undefined
  stack: |
    Object.<anonymous> (test/schema-feature.test.js:257:1)
    Module.replacementCompile (node_modules/append-transform/index.js:60:13)
    Object.<anonymous> (node_modules/append-transform/index.js:64:4)

​ FAIL ​ test/schema-feature.test.js
 ✖ child test left in queue: t.test Should emit warning if the schema body is undefined

​ FAIL ​ test/schema-feature.test.js
 ✖ child test left in queue: t.test Should emit warning if the schema query is undefined

​ FAIL ​ test/schema-feature.test.js
 ✖ child test left in queue: t.test Should emit warning if the schema params is undefined

​ FAIL ​ test/schema-feature.test.js
 ✖ child test left in queue: t.test Should emit a warning for every route with undefined schema

​ FAIL ​ test/schema-feature.test.js
 ✖ child test left in queue: t.test First level $ref

​ FAIL ​ test/schema-feature.test.js
 ✖ child test left in queue: t.test Customize validator compiler in instance and route

​ FAIL ​ test/schema-feature.test.js
 ✖ child test left in queue: t.test Use the same schema across multiple routes

​ FAIL ​ test/schema-feature.test.js
 ✖ child test left in queue: t.test Encapsulation should intervene

​ FAIL ​ test/schema-feature.test.js
 ✖ child test left in queue: t.test Encapsulation isolation

​ FAIL ​ test/schema-feature.test.js
 ✖ child test left in queue: t.test Add schema after register

​ FAIL ​ test/schema-feature.test.js
 ✖ child test left in queue: t.test Encapsulation isolation for getSchemas

​ FAIL ​ test/schema-feature.test.js
 ✖ child test left in queue: t.test Use the same schema id in different places

​ FAIL ​ test/schema-feature.test.js
 ✖ child test left in queue: t.test Get schema anyway should not add `properties` if allOf is present

​ FAIL ​ test/schema-feature.test.js
 ✖ child test left in queue: t.test Get schema anyway should not add `properties` if oneOf is present

​ FAIL ​ test/schema-feature.test.js
 ✖ child test left in queue: t.test Get schema anyway should not add `properties` if anyOf is present

​ FAIL ​ test/schema-feature.test.js
 ✖ child test left in queue: t.test Shared schema should be ignored in string enum

​ FAIL ​ test/schema-feature.test.js
 ✖ child test left in queue: t.test Shared schema should NOT be ignored in != string enum

​ FAIL ​ test/schema-feature.test.js
 ✖ child test left in queue: t.test Case insensitive header validation

​ FAIL ​ test/schema-feature.test.js
 ✖ child test left in queue: t.test Not evaluate json-schema $schema keyword

​ FAIL ​ test/schema-feature.test.js
 ✖ child test left in queue: t.test Validation context in validation result

​ FAIL ​ test/schema-feature.test.js
 ✖ child test left in queue: t.test The schema build should not modify the input

​ FAIL ​ test/schema-feature.test.js
 ✖ child test left in queue: t.test Cross schema reference with encapsulation references

​ FAIL ​ test/schema-feature.test.js
 ✖ child test left in queue: t.test Check how many AJV instances are built #1

​ FAIL ​ test/schema-feature.test.js
 ✖ child test left in queue: t.test onReady hook has the compilers ready

​ FAIL ​ test/schema-feature.test.js
 ✖ child test left in queue: t.test Check how many AJV instances are built #2 - verify validatorPool

​ FAIL ​ test/schema-feature.test.js
 ✖ child test left in queue: t.test Add schema order should not break the startup

​ FAIL ​ test/schema-feature.test.js
 ✖ child test left in queue: t.test The schema compiler recreate itself if needed

​ FAIL ​ test/schema-feature.test.js
 ✖ child test left in queue: t.test Schema controller setter

​ FAIL ​ test/schema-feature.test.js
 ✖ child test left in queue: t.test Schema controller bucket

​ FAIL ​ test/schema-feature.test.js
 ✖ child test left in queue: t.test setSchemaController per instance

​ FAIL ​ test/schema-feature.test.js
 ✖ child test left in queue: t.test setSchemaController: Inherits correctly parent schemas with a customized validator instance

​ FAIL ​ test/schema-feature.test.js
 ✖ child test left in queue: t.test setSchemaController: Inherits buildSerializer from parent if not present within the instance

​ FAIL ​ test/schema-feature.test.js
 ✖ child test left in queue: t.test setSchemaController: Inherits buildValidator from parent if not present within the instance

​ FAIL ​ test/schema-feature.test.js
 ✖ child test left in queue: t.test Should throw if not default validator passed

​ FAIL ​ test/schema-feature.test.js
 ✖ child test left in queue: t.test Should coerce the array if the default validator is used

​ FAIL ​ test/schema-feature.test.js
 ✖ child test left in queue: t.test Should return a human-friendly error if response status codes are not specified

​ FAIL ​ test/schema-feature.test.js
 ✖ child test left in queue: t.test setSchemaController: custom validator instance should not mutate headers schema

​ FAIL ​ test/schema-feature.test.js 38 failed of 76 688.167ms
 ✖ test unfinished
 ✖ child test left in queue: t.test Should emit warning if the schema body is undefined
 ✖ child test left in queue: t.test Should emit warning if the schema query is undefined
 ✖ child test left in queue: t.test Should emit warning if the schema params is undefined
 ✖ child test left in queue: t.test Should emit a warning for every route with undefined schema
 ✖ child test left in queue: t.test First level $ref
 ✖ child test left in queue: t.test Customize validator compiler in instance and route
 ✖ child test left in queue: t.test Use the same schema across multiple routes
 ✖ child test left in queue: t.test Encapsulation should intervene
 ✖ child test left in queue: t.test Encapsulation isolation
 ✖ child test left in queue: t.test Add schema after register
 ✖ child test left in queue: t.test Encapsulation isolation for getSchemas
 ✖ child test left in queue: t.test Use the same schema id in different places
 ✖ child test left in queue: t.test Get schema anyway should not add `properties` if allOf is present
 ✖ child test left in queue: t.test Get schema anyway should not add `properties` if oneOf is present
 ✖ child test left in queue: t.test Get schema anyway should not add `properties` if anyOf is present
 ✖ child test left in queue: t.test Shared schema should be ignored in string enum
 ✖ child test left in queue: t.test Shared schema should NOT be ignored in != string enum
 ✖ child test left in queue: t.test Case insensitive header validation
 ✖ child test left in queue: t.test Not evaluate json-schema $schema keyword
 ✖ child test left in queue: t.test Validation context in validation result
 ✖ child test left in queue: t.test The schema build should not modify the input
 ✖ child test left in queue: t.test Cross schema reference with encapsulation references
 ✖ child test left in queue: t.test Check how many AJV instances are built #1
 ✖ child test left in queue: t.test onReady hook has the compilers ready
 ✖ child test left in queue: t.test Check how many AJV instances are built #2 - verify validatorPool
 ✖ child test left in queue: t.test Add schema order should not break the startup
 ✖ child test left in queue: t.test The schema compiler recreate itself if needed
 ✖ child test left in queue: t.test Schema controller setter
 ✖ child test left in queue: t.test Schema controller bucket
 ✖ child test left in queue: t.test setSchemaController per instance
 ✖ child test left in queue: t.test setSchemaController: Inherits correctly parent schemas with a customized validator instance
 ✖ child test left in queue: t.test setSchemaController: Inherits buildSerializer from parent if not present within the instance
 ✖ child test left in queue: t.test setSchemaController: Inherits buildValidator from parent if not present within the instance
 ✖ child test left in queue: t.test Should throw if not default validator passed
 ✖ child test left in queue: t.test Should coerce the array if the default validator is used
 ✖ child test left in queue: t.test Should return a human-friendly error if response status codes are not specified
 ✖ child test left in queue: t.test setSchemaController: custom validator instance should not mutate headers schema
```

Reply:
With the suggestion I made?
So far, seems that only FSTWRN001 is in use

Reply:
Yep. Looks like there was an issue with the order. For now I put `FSTWRN002` above `FSTWRN001`.